### ISSUE TITLE
salt.gitfs.gitpython state assumes pillar data will exist

### ISSUE THREAD

**Issue by colin-stubbs** [saltstack-formulas/salt-formula#361]
State assumes pillar data will exist. Should instead using defaults and/or os/os_family map to generate defaults to ensure a sane default exists within referenced pillar.

Example of issue,

```
[root@box ~]# salt 'box' state.highstate
box:
    Data failed to compile:
----------
    Rendering SLS 'base:salt.gitfs.gitpython' failed: Jinja variable 'dict object' has no attribute 'gitpython'


-------------------------------------------
Summary
-------------------------------------------
# of minions targeted: 1
# of minions returned: 1
# of minions that did not return: 0
# of minions with errors: 1
-------------------------------------------
ERROR: Minions returned with non-zero exit code
[root@box ~]#
```

**Comment by aboe76** [comment]
@colin-stubbs can you test with the current pr #395

**Comment by aboe76** [comment]
fixed with #395
